compiler/src/dotty/tools/dotc/config/Settings.scala

```diff
@@ -88,7 +88,16 @@ object Settings {
         state.update(idx, x.asInstanceOf[T])
     }
 
-    def isDefaultIn(state: SettingsState) = valueIn(state) == default
+    def isDefaultIn(state: SettingsState): Boolean = valueIn(state) == default
+
+    def defaultValue: String = implicitly[ClassTag[T]] match {
```

Comment:
I would make it an inner method of `def helpStr`. Also you don't need a cast here:
```scala
def defaultValue(setting: Setting[_]): String = setting.default match {
  case _: Int | _: String => setting.default.toString
  case _ => ""
}
```